lib/eventum/auth/APIAuthToken.php

```diff
@@ -0,0 +1,159 @@
+
+        return $token;
+    }
+
+    public static function saveToken($usr_id, $token)
+    {
+        $sql = 'INSERT INTO
+                    {{%api_token}}
+                SET
+                    apt_usr_id = ?,
+                    apt_created = ?,
+                    apt_token = ?';
+        try {
+            $res = DB_Helper::getInstance()->query($sql, array(
+                $usr_id,
+                Date_Helper::getCurrentDateGMT(),
+                $token,
+            ));
+        } catch (DbException $e) {
+            return -1;
+        }
+    }
+
+    public static function isTokenValidForEmail($token, $email)
+    {
+        try {
+            if (self::checkTokenStatus($token) == User::getUserIDByEmail($email, true)) {
+                return true;
+            }
+        } catch (AuthException $e) {
+            return false;
+        }
+    }
+
+    public static function checkTokenStatus($token)
+    {
+        $sql = "SELECT
+                    apt_usr_id
+                FROM
+                    {{%api_token}}
+                WHERE
+                    apt_token = ? AND
+                    apt_status = 'active'";
+        try {
+            $usr_id = DB_Helper::getInstance()->getOne($sql, array($token));
+        } catch (DbException $e) {
+            throw new AuthException('Error fetching user token');
+        }
+        if (empty($usr_id)) {
+            throw new AuthException('Invalid token');
+        }
+
+        return $usr_id;
+    }
+
+    public static function getUserIDByToken($token)
+    {
+        try {
+            // check that token is valid and not expired
+            $decoded = JWT::decode($token, Auth::privateKey(), array(self::$default_alg));
+
+            // make sure token hasn't been revoked
+            $usr_id = self::checkTokenStatus($token);
+
+            return $usr_id;
+        } catch (Exception $e) {
+            throw new AuthException($e);
+        }
+    }
+
+    public static function getTokensForUser($usr_id, $auto_generate = false)
+    {
+        $sql = 'SELECT
+                    apt_id,
+                    apt_usr_id as usr_id,
+                    apt_created as created,
+                    apt_status as status,
+                    apt_token as token
+                FROM
+                    {{%api_token}}
+                WHERE
+                    apt_usr_id = ?
+                ORDER BY
+                    apt_created DESC';
+        try {
+            $res = DB_Helper::getInstance()->getAll($sql, array($usr_id));
+        } catch (DbException $e) {
+            return array();
+        }
+        if (empty($res)) {
+            if ($auto_generate) {
+                self::generate($usr_id);
+
+                return self::getTokensForUser($usr_id, false);
+            } else {
+                return array();
+            }
+        }
+
+        return $res;
+    }
+
+    public static function regenerateKey($usr_id)
```

Comment:
I don't think we need multiple active tokens, at least not until we use tokens for more then just the CLI.

The point of having token status is so we can invalidate old tokens but still have them for reference if needed.